src/python/pants/backend/python/lint/bandit/subsystem.py

```diff
@@ -79,6 +80,24 @@ def config_request(self) -> ConfigFilesRequest:
         )
 
 
+@rule_helper
+async def _bandit_interpreter_constraints(python_setup: PythonSetup) -> InterpreterConstraints:
+    # While Bandit will run in partitions, we need a set of constraints that works with every
+    # partition.
+    #
+    # This ORs all unique interpreter constraints. The net effect is that every possible Python
+    # interpreter used will be covered.
+    all_tgts = await Get(AllTargets, AllTargetsRequest())
+    unique_constraints = {
+        InterpreterConstraints.create_from_targets([tgt], python_setup)
+        for tgt in all_tgts
+        if BanditFieldSet.is_applicable(tgt)
+    }
+    constraints = InterpreterConstraints(itertools.chain.from_iterable(unique_constraints))
+
```

Comment:
Nit: unnecessary white space imo since these lines are highly related. There are a couple other places in the PR like this. Feel free to ignore, not a very big deal